File: Python/groupAnagrams.py
from collections import defaultdict
# from collections import Counter
def groupAnagrams(strs):
    """
    :type strs: List[str]
    :rtype: List[List[str]]
    """
    res = defaultdict(list)

    for s in strs:
        ordered_s = ''.join(sorted(s))
        print(ordered_s, s, res)
        res[ordered_s].append(s)
    return res.values()


print(groupAnagrams(["eat","tea","tan","ate","nat","bat"]))

File: groupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let res = new Map();
    strs.forEach((word) => {
        let sortedWord = word.split("").sort().join("")
        if(res.has(sortedWord)){
            res.get(sortedWord).push(word)
        } else {
            res.set(sortedWord, [word])
        }
    })
    let ans = []
    res.forEach((value) => {
        ans.push(value)
    })
    return ans
};


console.log(groupAnagrams(["eat","tea","tan","ate","nat","bat"]))